File: src/main/java/net/atos/zac/policy/input/DocumentInput.java
/*
 * SPDX-FileCopyrightText: 2022 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.zac.policy.input;

import jakarta.json.bind.annotation.JsonbProperty;

import nl.info.zac.authentication.LoggedInUser;

public class DocumentInput extends UserInput {

    @JsonbProperty("document")
    private final DocumentData documentData;

    public DocumentInput(final LoggedInUser loggedInUser, final DocumentData documentData) {
        super(loggedInUser);
        this.documentData = documentData;
    }

    public DocumentData getDocumentData() {
        return documentData;
    }
}

File: src/main/kotlin/nl/info/zac/policy/input/DocumentInput.kt
/*
 * SPDX-FileCopyrightText: 2022 Atos, 2025 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package nl.info.zac.policy.input

import jakarta.json.bind.annotation.JsonbProperty
import nl.info.zac.authentication.LoggedInUser
import nl.info.zac.util.AllOpen
import nl.info.zac.util.NoArgConstructor

@NoArgConstructor
@AllOpen
data class DocumentInput(
    var loggedInUser: LoggedInUser,

    @field:JsonbProperty("document")
    var documentData: DocumentData
) : UserInput(loggedInUser)
